src/main/java/com/ace/alfox/game/MoveAction.java

```diff
@@ -33,20 +37,43 @@ public MoveAction(Database db) {
   @Override
   public ActionResult applyAction(Player player, Map<String, Object> params) {
     ActionResult result = new ActionResult(player);
+    if (!player.canMove()) {
+      result.log("Cannot move.");
+
+      if (player.inBattle()) {
+        result.log(getBattleLog(this.db.battles.find(player), db.locations.find(player.location)));
+      }
+
+      return result;
+    }
+
     String direction = (String) params.get("direction");
 
     if (player.hp == 0) return result.notOk().log("You're dead.");
-    if (direction == null || !directions.containsKey(direction))
+    if (direction == null || !directions.containsKey(direction)) {
       return result.notOk().log("Unknown direction"); // notice .log returns `ActionResult`
+    }
 
     // player.attacking.clear() // if they were attacking anyone, they aren't anymore
     // maybe do running away mechanic here?
     player.location = player.location.add(directions.get(direction));
     result.log("traveled " + direction);
 
     Location location = db.locations.find(player.location);
-    result.log("You see, " + location.title).log(location.description);
+    if (random.nextInt(5) == 0) {
```

Comment:
haha, I was wondering how you were going to do it.

Reply:
In Shimlar you picked when to enter battles. Not sure what we wanted to do for this game, so I went with random battles since it was easy